lib/bundler/source.rb

```diff
@@ -38,5 +38,17 @@ def include?(other)
     def inspect
       "#<#{self.class}:0x#{object_id} #{self}>"
     end
+
+  private
+
+    def version_color(spec_version, locked_spec_version)
+      earlier_version?(spec_version, locked_spec_version) ? :yellow : :green
+    end
+
+    def earlier_version?(spec_version, locked_spec_version)
+      Gem::Version.new(spec_version) < Gem::Version.new(locked_spec_version)
+    rescue ArgumentError
```

Comment:
@segiddins I can confirm that despite being named `spec.version`, the `version_message(spec)` method does get passed things that are not proper versions. For instance, in running `spec/bundler/source_spec.rb`, it frequently gets passed in an object for which `spec.version` is `">= 1.6"`.

I didn’t know about `Gem::Version.correct?` but that is much better than catching an exception, I totally agree.

Added a commit here that made that change.

This leaves two issues remaining:

1. Ideally we only want to even call `earlier_version?` when comparing actual concrete version numbers. I’m still learning the Bundler codebase and don’t know offhand how I would do this.

2. We need some specs to positively assert cases where we want/expect the amber color to be present. Again, I’m not very familiar with the Bundler code so I would love any help here if possible.